include/arch/arm/arch.h

```diff
@@ -153,6 +153,23 @@ extern "C" {
 		sym[size+MPU_GUARD_ALIGN_AND_SIZE]
 #endif
 
+/**
+ * @brief Calculate size of stacks to be allocated in a stack array
```

Comment:
Why are you documenting the internal `_ARCH_THREAD_STACK_LEN` with a doxygen block?  Alternatively, why does only ARM get the docs?  This seems like mostly a duplication of the K_* macro below, so I'd remove it.  Or if you think it's needed for arch authors, maybe move it to the porting guide?

Reply:
I have no opinion on this. As you pointed out, it's there only for ARM and I copied it for uniformity purposes.